### PR TITLE
Quarterdeck accept invite

### PR DESCRIPTION
**Pull request by pdeziel**
### Scope of changes

This adds an authenticated endpoint to Quarterdeck for accepting user invites and refactors some logic into a shared helper method.

Fixes SC-20702

### Type of change

- [x] new feature
- [ ] bug fix
- [ ] documentation
- [ ] testing
- [ ] technical debt
- [ ] other (describe)

### Acceptance criteria

Port the accept invitation code from the login endpoint into an accept invitation endpoint. On invalid invitation this should return 400 and on valid invitation the user should be put on the organization.

### Definition of Done

- [x] I have manually tested the change running it locally (having rebuilt all containers) or via unit tests 
- [x] I have added unit and/or integration tests that cover my changes
- [ ] I have added new test fixtures as needed to support added tests
- [ ] I have updated the dependencies list if necessary (including updating yarn.lock and/or go.sum)
- [ ] I have recompiled and included new protocol buffers to reflect changes I made if necessary
- [x] Check this box if a reviewer can merge this pull request after approval (leave it unchecked if you want to do it yourself)
- [x] I have notified the reviewer via Shortcut or Slack that this is ready for review
- [ ] Front-end: Checked sm, md, lg screen resolutions for effective responsiveness
- [ ] Backend-end: Documented service configuration changes or created related devops stories

### Reviewer(s) checklist

- [ ] Front-end: I've reviewed the Figma design and confirmed that changes match the spec.
- [ ] Any new user-facing content that has been added for this PR has been QA'ed to ensure correct grammar, spelling, and understandability.
- [ ] Are there any TODOs in this PR that should be turned into stories?

### CODE REVIEW

File: pkg/quarterdeck/invites.go [hunk 1]
```diff
@@ -198,3 +202,140 @@ func (s *Server) InviteCreate(c *gin.Context) {
...
+		c.JSON(http.StatusInternalServerError, api.ErrorResponse(responses.ErrSomethingWentWrong))
+		return err
+	}
+
+	// Verify that the invite is for this email address
+	if err = invite.Validate(user.Email); err != nil {
+		switch {
+		case errors.Is(err, db.ErrTokenExpired):
+			log.Debug().Msg("invite token expired")
+			metrics.FailedLogins.WithLabelValues(ServiceName, UserHuman, "invite token expired").Inc()
```
Comment (pdeziel) [inline pkg/quarterdeck/invites.go:302]: Not sure if this is going to break the metrics, these strings are a bit different than they were in Login.

**Review by bbengfort — APPROVED** [review]
This was very tough to review -- I didn't see anything concerning but I think it would be good if you, me, and @pdamodaran could step through each line of code to make sure we're all on the same page with what's happening during office hours?

File: pkg/quarterdeck/api/v1/client.go [hunk 2]
```diff
@@ -485,6 +489,21 @@ func (s *APIv1) InviteCreate(ctx context.Context, in *UserInviteRequest) (out *U
 	return out, nil
 }
 
+func (s *APIv1) InviteAccept(ctx context.Context, token string) (out *LoginReply, err error) {
+	endpoint := fmt.Sprintf("/v1/invites/%s", token)
```
Comment (bbengfort) [inline pkg/quarterdeck/api/v1/client.go:493]: It's not typical that you POST `nil` data to an endpoint; the token is data not an identity. Is it possible to do `POST /v1/invites` and put the token in the body?

Comment (bbengfort) [reply pkg/quarterdeck/api/v1/client.go:493]: Or alternatively do a POST to `/v1/invites/:userID`? The client could then parse the userID out of the token.

**Pull request merged**